enumerable.rb

```diff
@@ -0,0 +1,124 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    my_array = is_a?(Range) ? to_a : self
+
+    my_array.size.times { |n| yield (my_array[n]) }
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    my_array = is_a?(Range) ? to_a : self
+
+    my_array.size.times { |n| yield my_array[n], n }
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    result_array = []
+    my_each { |n| result_array << n if yield(n) }
+    result_array
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def my_all(arg = nil)
+    result = true
+
+    my_each do |i|
```

Comment:
### #my_all? 
* [ ] All Boolean function should end with a question mark for best practices

Reply:
Fixed